File: cpp/svg/common/svgDwg.cpp
// svgDwg.cpp

using namespace std;

#include "svgDwg.h"
#include "xml.h"
#include <string> 
#include <stdio.h>
                                    

svgDwg::svgDwg( float width , float height )
: _width( width )
, _height( height )
{
}

void svgDwg::start( void ) 
{
  char buff[100] ;

  sprintf( buff , "0 0 %d %d" , (int)( _width * 100 ) ,(int)( _height * 100 ) ) ;

  xmlWrite( "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>" ) ;
  xmlStartMultitag( "svg" ) ;
  xmlWritePar( "xmlns:svg" , "http://www.w3.org/2000/svg" ) ;
  xmlWritePar( "xmlns" , "http://www.w3.org/2000/svg" ) ;
  xmlWritePar( "version" , "1.1" ) ;
  xmlWritePar( "width" , _width   , "in" ) ;
  xmlWritePar( "height" , _height , "in" ) ;
  xmlWritePar( "viewBox" , buff ) ;
  xmlPartialendMultitag() ;
}

void svgDwg::end( void )
{
  xmlEndTag( "svg" ) ;
}

File: python/svg/svgDWG.py
import XML as xml
from svgGROUP import svgGroup

class svgDwg( svgGroup ):
  
  def __init__( self , width = 1 , height = 1 ):
    self._width = width
    self._height = height
    super(svgDwg,self).__init__()

  def draw(self):
    self.Start()
    super(svgDwg,self).draw() 
    self.End()
        
    
  def Start(self):
    buff = "0 0 %d %d" % ( self._width ,  self._height )
    xml.Write( "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>" ) 
    xml.StartMultitag( "svg" ) 
    xml.WritePar( "xmlns:svg" , "http://www.w3.org/2000/svg" ) 
    xml.WritePar( "xmlns" , "http://www.w3.org/2000/svg" ) 
    xml.WritePar( "version" , "1.1" ) 
    xml.WritePar( "width" , self._width   , "px" ) 
    xml.WritePar( "height" , self._height , "px" ) 
    xml.WritePar( "viewBox" , buff ) ;
    xml.PartialendMultitag() ;
  
  def End(self):
    xml.EndTag( "svg" ) ;
    